dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AuthenticatedResourceInterface.java

```diff
@@ -15,266 +15,181 @@
-     *
-     * @param user the user that is requesting something
-     * @param entry entry to check permissions for()
+     * Check if a specified token exists (is not null).
+     * @param token token to be checked
      */
-    default void checkUserCanShare(User user, Entry entry) {
-        checkUser(user, entry);
+    default void checkExistsToken(Token token) {
+        throwIf(token == null, "Token not found.", HttpStatus.SC_NOT_FOUND);
     }
 
-    /**
-     * Override for resources that are permissions aware.
-     * Currently done for WorkflowResource
-     * @param user the user that is requesting something
-     * @param entry entry to check permissions for()
-     */
-    default void checkCanRead(User user, Entry entry) {
-        throw new CustomWebApplicationException("Forbidden: you do not have the credentials required to access this entry.",
-            HttpStatus.SC_FORBIDDEN);
+    default boolean canRead(User user, Entry<?, ?> entry) {
+        return (entry != null && entry.getIsPublished()) || isOwner(user, entry);
     }
 
-    /**
-     * This method checks that a workflow can be read in two situations
-     * 1) A published workflow
-     * 2) A workflow that is unpublished but that I have access to
-     *
-     * @param user the user that is requesting something
-     * @param entry entry to check permissions for()
-     */
-    default void checkOptionalAuthRead(Optional<User> user, Entry entry) {
-        checkEntry(entry);
-        if (!entry.getIsPublished()) {
-            if (user.isPresent()) {
-                checkCanRead(user.get(), entry);
-            } else {
-                throw new CustomWebApplicationException("Forbidden: you do not have the credentials required to access this entry.",
-                    HttpStatus.SC_FORBIDDEN);
-            }
-        }
+    default boolean canWrite(User user, Entry<?, ?> entry) {
+        return isOwner(user, entry);
     }
 
-    /**
-     * Check if organization is null
-     *
-     * @param organization organization to check permissions for
-     */
-    default void checkOrganization(Organization organization) {
-        if (organization == null) {
-            throw new CustomWebApplicationException("Organization not found", HttpStatus.SC_NOT_FOUND);
-        }
-
+    default boolean canShare(User user, Entry<?, ?> entry) {
+        return isOwner(user, entry);
     }
 
-    /**
-     * Only passes if entry is published or if user has correct credentials
-     * @param user Optional user
-     * @param entry Entry to check
-     */
-    default void optionalUserCheckEntry(Optional<User> user, Entry entry) {
-        if (!entry.getIsPublished()) {
-            if (user.isEmpty()) {
-                throw new CustomWebApplicationException("Entry not found", HttpStatus.SC_BAD_REQUEST);
-            } else {
-                checkUser(user.get(), entry);
-            }
-        }
+    default boolean isOwner(User user, Entry<?, ?> entry) {
```

Comment:
What do you think about making this and other `can...` methods, e.g., `canShare`, static? You seem to essentially be using them like private methods -- they're never overriden; the pattern seems to be to instead override the `check..` methods, e.g., `checkCanShare`. You're constrained because this is an interface, not a class, and making the static is sort of lame, but at least it removes the expectation that the method will be overriden, making it somewhat easier to read.

I was confused at first reading the code until I realized that. If you disagree, then perhaps at least some JavaDoc on which methods are expected to overridden and which ones aren't.

Reply:
In the new implementation, the "can" methods are intended to be overridden to implement additional auth methods, and the "check" methods wrap the "can" methods.  I added a little blurb to that effect in the javadoc comment for `AuthenticatedResourceInterface`.